Content.Client/Construction/UI/ConstructionMenuPresenter.cs

```diff
@@ -185,40 +205,50 @@ private void OnViewPopulateRecipes(object? sender, (string search, string catago
             // There is apparently no way to set which
         }
 
-        private void PopulateCategories()
+        private void PopulateCategories(string? selectCategory = null)
         {
             var uniqueCategories = new HashSet<string>();
 
-            // hard-coded to show all recipes
-            uniqueCategories.Add("construction-category-all");
-
             foreach (var prototype in _prototypeManager.EnumeratePrototypes<ConstructionPrototype>())
             {
                 var category = prototype.Category;
 
                 if (!string.IsNullOrEmpty(category))
                     uniqueCategories.Add(category);
             }
+            var sortedCategories = uniqueCategories.OrderBy(x => Loc.GetString(x)).ToList();
+            // hard-coded to show all recipes
+            sortedCategories.Insert(0, _forAllCategoryName);
 
```

Comment:
This is going to shift the entire list along which is incredibly slow, just allocate sortedcategories up front (you already have its size) then add _forAllCategoryName then add the localised string afterwards.

Reply:
I don't know, this item was already there, and I didn't touch it. But I'll try to remove it.

Reply:
I removed the insertions of elements into the lists and tried to simplify this code. Maybe you had something else in mind, but this is what I came up with.